dessia_common/datatools/modeling.py

```diff
@@ -0,0 +1,734 @@
+
+    @classmethod
+    def _instantiate_dessia(cls, model, name: str = ''):
+        raise NotImplementedError(f'Method _instantiate_dessia not implemented for {cls.__name__}.')
+
+    @classmethod
+    def fit_(cls, inputs: List[List[float]], outputs: List[List[float]], name: str = '', **hyperparameters):
+        model = cls._skl_class()(**hyperparameters)
+        model.fit(inputs, outputs)
+        return cls._instantiate_dessia(model, name)
+
+    def predict(self, inputs: List[List[float]]):
+        model = self._instantiate_skl()
+        return model.predict(inputs).tolist()
+
+    @classmethod
+    def fit_predict_(cls, inputs: List[List[float]], outputs: List[List[float]], predicted_inputs: List[List[float]],
+                    name: str = '', **hyperparameters):
+        model = cls.fit_(inputs, outputs, name, **hyperparameters)
+        return model, model.predict(predicted_inputs)
+
+    def score(self, inputs: List[List[float]], outputs: List[List[float]]):
+        model = self._instantiate_skl()
+        return model.score(inputs, outputs)
+
+
+class LinearRegression(BaseModel):
+    _standalone_in_db = True
+
+    def __init__(self, coef_: List[List[float]] = None, intercept_: List[List[float]] = None, name: str = ''):
+        self.coef_ = coef_
+        self.intercept_ = intercept_
+        BaseModel.__init__(self, name=name)
+
+    @classmethod
+    def _skl_class(cls):
+        return linear_model.Ridge
+
+    def _instantiate_skl(self):
+        model = self._call_skl_model()
+        model.coef_ = npy.array(self.coef_)
+        model.intercept_  = npy.array(self.intercept_)
+        return model
+
+    @classmethod
+    def _instantiate_dessia(cls, model, name: str = ''):
+        return cls(coef_=model.coef_.tolist(), intercept_=model.intercept_.tolist(), name=name)
+
+    @classmethod
+    def fit(cls, inputs: List[List[float]], outputs: List[List[float]], name: str = '',
+            alpha: float = 1., fit_intercept: bool = True, tol: float = 0.001):
+        return cls.fit_(inputs, outputs, name=name, alpha=alpha, fit_intercept=fit_intercept, tol=tol)
+
+    @classmethod
+    def fit_predict(cls, inputs: List[List[float]], outputs: List[List[float]], predicted_inputs: List[List[float]],
+                    name: str = '', alpha: float = 1., fit_intercept: bool = True, tol: float = 0.001):
+        return cls.fit_predict_(inputs, outputs, predicted_inputs, name=name,
+                                alpha=alpha, fit_intercept=fit_intercept, tol=tol)
+
+
+class BaseTree(BaseModel):
+
+    def __init__(self, n_classes: List[int] = None, n_features: int = None, n_outputs: int = None,
+                 tree_state: Dict[str, Any] = None, name: str = ''):
+        self.n_classes = n_classes
+        self.n_features = n_features
+        self.n_outputs = n_outputs
+        self.tree_state = tree_state
+        BaseModel.__init__(self, name=name)
+
+    def _data_hash(self):
```

Comment:
Red light is turning on here ! data_hash is overwritten but not data_eq. Why is that ?

It might be perfectly all right, but I feel obliged to ask because that might be "dangerous" as well

Reply:
Actually, i did not want to rewrite anything of these methods but i have encountered some performance problems when calling check_platform method for RandomForest.

RandomForest is a list of DecisionTree which contain BaseTree.
So, when running the check_platform, RandomForest could take some 45 seconds to be checked (n_estimators_  x DecisionTree, each containing a BaseTree). I think it is too long. So I made a profiling, which sent me to the data_hash method for BaseTree.

After some investigations, it seems like the data_hash method does not like so much dict of list of list, from the tree_state attribute. 
So I just changed it dumbly without questioning anything else, counting on you to correct me if i'm wrong.

Do you suggest me to also rewrite the data_eq method for BaseTree ?
Why ?

Reply:
So if i have `a._data_hash() == b._data_hash()` the next step is necessarily to check `a.__eq__(b)`, isn't it ?

If yes, what i wrote is ok for your warning